Add blog post about Workload Identity (#958)
gardener/documentation: website/blog/2026/05/05-05-workload-identity-eliminate-static-credentials-for-your-gardener-shoot-clusters.md

- **Broad permissions** — credentials are often over-provisioned because tightening them requires careful auditing.
- **Storage risks** — secrets stored in multiple locations (CI/CD pipelines, Secret managers, Kubernetes clusters) multiply the attack surface.
- **Expired credentials break reconciliation** — if a credential silently expires, cluster operations stall until someone notices and fixes it.

## How Workload Identity Solves This

With workload identity, Gardener acts as an **OIDC-compatible token issuer**. Instead of handing Gardener a static secret, you establish a one-time trust relationship between your cloud provider account and the Gardener issuer. From that point on:

1. **Gardener issues short-lived JWTs** — tokens are created on-demand by the Gardener API server and contain claims identifying the specific shoot, project, and seed.
1. **Tokens are auto-rotated** — gardenlet continuously refreshes tokens before they expire, with no manual intervention.
1. **No credentials are stored** — tokens are ephemeral and never persisted by the issuer. The cloud provider validates them using the publicly available OIDC discovery metadata.

> 1. **Fine-grained access control** — cloud-side trust policies can restrict which Gardener workload identities (by subject claim) are allowed to assume which roles, enabling per-shoot or per-project isolation.

Reviewer: ```suggestion
1. **Fine-grained access control** — cloud-side trust policies can restrict Gardener workload identities (by subject claim) to concrete IAM roles, enabling per-shoot or per-project isolation.
```

If your shoot references a `SecretBinding`, create a `CredentialsBinding` that points to the **same Secret** and update the shoot:

```yaml
apiVersion: security.gardener.cloud/v1alpha1
kind: CredentialsBinding
metadata:
  name: my-infra-credentials
  namespace: garden-myproject
credentialsRef:
  apiVersion: v1
  kind: Secret
  name: my-infra-secret
  namespace: garden-myproject
provider:
  type: aws  # or "azure" or "gcp"
```

Then replace `.spec.secretBindingName` with `.spec.credentialsBindingName` in your Shoot:

```yaml
spec:
  credentialsBindingName: my-infra-credentials  # was: secretBindingName: my-old-binding
```

> [!NOTE]
> This step requires that the new `CredentialsBinding` references the exact same `Secret` as the old `SecretBinding`.
> A direct migration from `SecretBinding` to a `CredentialsBinding` referencing a `WorkloadIdentity` is not allowed.

For detailed instructions, see the [SecretBinding to CredentialsBinding migration guide](https://github.com/gardener/gardener/blob/master/docs/usage/shoot-operations/secretbinding-to-credentialsbinding-migration.md).

### Step 2: Switch to Workload Identity

> Once your shoot uses a `CredentialsBinding` with a `Secret`, you can move to `WorkloadIdentity`:

Reviewer: ```suggestion
Once your shoot is using a `CredentialsBinding` with a `Secret`, you can move to `WorkloadIdentity`:
```

1. Set up the trust relationship in your cloud provider account (see [Provider-Specific Setup](#provider-specific-setup) above).
1. Create a `WorkloadIdentity` resource in your project namespace.
1. Create a **new** `CredentialsBinding` that references the `WorkloadIdentity`:

    ```yaml
    apiVersion: security.gardener.cloud/v1alpha1
    kind: CredentialsBinding
    metadata:
      name: my-infra-wi-binding
      namespace: garden-myproject
    credentialsRef:
      apiVersion: security.gardener.cloud/v1alpha1
      kind: WorkloadIdentity
      name: my-infra-identity
    provider:
      type: aws  # or "azure" or "gcp"
    ```

1. Update your Shoot's `.spec.credentialsBindingName` to point to the new binding:

    ```yaml
    spec:
      credentialsBindingName: my-infra-wi-binding
    ```

> Since the `credentialsRef` field of `CredentialsBinding` is immutable, you create a new `CredentialsBinding` rather than modifying the existing one. The migration is seamless — Gardener handles the transition without cluster downtime.

Reviewer: ```suggestion
Since the `credentialsRef` field of `CredentialsBinding` is immutable, you must create a new `CredentialsBinding` rather than modifying the existing one. The migration is seamless — Gardener handles the transition without cluster downtime.
```
